File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let cnt = 0;
    let day = [];
    let count = 1;
    
    // 각각의 작업이 걸리는 일수 계산
    for(let i = 0; i < progresses.length; i++) {
      while((progresses[i] - 100) < 0){
        progresses[i] += speeds[i];
        cnt++;
      }
      day.push(cnt);
      cnt = 0;
    }
    //console.log("1day:",day) // 	day: [ 7, 3, 9 ] day: [ 5, 10, 1, 1, 20, 1 ]
    
    let maxDay = day[0];
    // 작업당 걸리는 배포 일수 계산
    for(let i=1; i<day.length; i++) {
        if(maxDay >= day[i]){
            count++;
        } else {
            maxDay = day[i];
            answer.push(count);
            count = 1;
        }
    }
    answer.push(count);
    return answer;
}

File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> result = new ArrayList<>();
        int[] days = new int[progresses.length];
        
        // 작업 일수
        for(int i=0; i<progresses.length; i++) {
            while(progresses[i] < 100) {
                progresses[i] += speeds[i];
                days[i]++;
            }
        }
        
        int max = days[0];
        int count = 1;
        // 배포 일수
        for(int i=1; i<days.length; i++) {
            if(max >= days[i]){
                count++;
            } else {
                max = days[i];
                result.add(count);
                count = 1;
            }
        }
        result.add(count); // 마지막 배포 일수 추가
        
        int[] answer = result.stream().mapToInt(Integer::intValue).toArray();

        return answer;
    }
}